rcl/src/rcl/guard_condition.c

```diff
@@ -119,7 +122,7 @@ rcl_guard_condition_fini(rcl_guard_condition_t * guard_condition)
   if (guard_condition->impl) {
     // assuming the allocator is valid because it is checked in rcl_guard_condition_init()
     rcl_allocator_t allocator = guard_condition->impl->options.allocator;
-    if (guard_condition->impl->rmw_handle) {
+    if (guard_condition->impl->rmw_handle && guard_condition->impl->allocated_rmw_guard_condition) {
```

Comment:
Huh, interesting, linters seem OK with the change.  It is exactly 100 characters long, which I think is the absolute allowed length, so I got lucky :).